platform/stm32f7xx/qspi.c

```diff
@@ -75,14 +81,36 @@ static uint32_t get_address_size(uint32_t address);
 static event_t cmd_event;
 static event_t rx_event;
 static event_t tx_event;
+static event_t st_event;
 
 status_t hal_error_to_status(HAL_StatusTypeDef hal_status);
 
+static status_t dma_disable(DMA_Stream_TypeDef *dma)
+{
+    // Unset the DMA Enable bit.
+    dma->CR &= ~DMA_SxCR_EN;
+
+    uint attempts = 0;
+
+    while (dma->CR & DMA_SxCR_EN) {
+
+        thread_sleep(1);
```

Comment:
That's pretty grody, is there a better way to spin on a in-progress bit and wait for that to clear? Is dma_disable used in the common path? If so, this thread sleep will definitely slow things down.